### ISSUE TITLE
Mopidy: Library view is empty

### ISSUE THREAD

**Issue by gvc0** [CDrummond/cantata#949]
Cantata (2.0.1) on Ubuntu won't load the library from my Mopidy server. Songs are available via the Folders and Search views. I tried the 'Refresh database' menu option, and also deleted the local .sql file to no avail.

When running Cantata in debug mode 1 I notice it issues an MPD "stats" command when executing a manual 'Refresh database'. As Mopidy always returns all zero's for this command it seems that Cantata concludes an update is not required. As a result it's impossible to populate the Library view.

If my analysis is correct, wouldn't it be possible to provide a 'force manual update' method, which doesn't perform the stats check and always loads the database? If I'm not mistaken this option was available with an older version I used on my Windows PC, which worked fine with Mopidy.

MPD debug info:
`MPDConnection 139875743430400 0x231aba0 sendCommand - sent 
MPDConnection 139875743430400 0x231aba0 sendCommand: "stats" true true 
MPDConnection 139875743430400 Timeout (ms): 2000 
MPDConnection 139875743430400 Socket state after write: 3 
MPDConnection 139875743430400 0x231aba0 Waiting for read data, attempt 0 
MPDConnection 139875743430400 0x231aba0 Read: "uptime: 0
db_update: 0
artists: 0
playtime: 0
albums: 0
db_playtime: 0
songs: 0
OK
`

**Comment by CDrummond** [comment]
This is a Mopdiy issue - it does not support the full MPD command set. Cantata used to have Mopidy workarounds, but these were removed as Mopdiy changed/removed support for certain commands.

Cantata is an **MPD** client,

**Comment by gvc0** [comment]
I agree it's a Mopidy issue. But providing a forced (full) database refresh option is a pretty easy workaround I think, which might even come in handy in problem situations with other MPD servers.

I don't think Mopidy changed or removed MPD commands. It has disabled the 'listallinfo' command by default, but you can easily reenable it in the configuration file ('command_blacklist' value).

From a usability perspective I think Cantata is by far the best MPD client for Windows and Linux, however all other MPD clients do work with Mopidy. So although it is a Mopidy issue Canata has some design choices that prevents these very nice products working together.

**Comment by CDrummond** [comment]
This should now be fixed in master (for 2.1.0) - at least it is for me. Can you checkout git/master and confirm?

**Comment by gvc0** [comment]
Thanks for picking this up. I still didn't succeed in building the application (on Ubuntu) due to a compliation error in models/**browsemodel.cpp**, line 256:
`PlayQueueModel::encode(*mimeData, PlayQueueModel::constFileNameMimeType, files);`

Error: 'encode' is not a member of 'PlayQueueModel'
Error: 'constFileNameMimeType' is not a member of 'PlayQueueModel'

**Comment by CDrummond** [comment]
Ah, dont use the -DENABLE_UBUNTU flag. This is for a very old (and no longer compiling) Ubuntu Touch version. I need to remove this flag, etc.

Just call make as follows:

1. mkdir build
2. cmake <path to source folder> -DCMAKE_INSTALL_PREFIX=/usr
3. make && sudo make install

**Comment by gvc0** [comment]
That worked out fine, however now I've got an error during compilation of 'Translations':
`lconvert: could not exec '/usr/lib/x86_64-linux-gnu/**qt4**/bin/lconvert': No such file or directory` (full log in attachment).
This refers to QT4, which I didn't install because I thought the new versions of Cantata only require QT5? (I even tried using the -DENABLE_QT5=ON option).

Anyway, I worked around that and compiled the code. I'm happy to confirm the Cantata library and database refresh works fine again with Mopidy. Thanks for you support!

[Compilation error.txt](https://github.com/CDrummond/cantata/files/701107/Compilation.error.txt)
